yapapi/engine.py

```diff
@@ -448,26 +455,48 @@ async def _process_invoices(self) -> None:
             ):
                 break
 
+    async def _check_debit_note_rate(self, act_id: ActivityId, agr_id: AgreementId, job: "Job"):
+        agreement = self._get_agreement_by_id(agr_id)
+        self._number_of_debit_notes[act_id] += 1
```

Comment:
This **definitely** shouldn't be done here:
1. https://en.wikipedia.org/wiki/Command%E2%80%93query_separation
2. "Check" means "check", not "update state and than check"
3. This assumes we call this function exactly once per debit note

Reply:
Fixed